streams/src/main/java/org/apache/kafka/streams/kstream/internals/CacheEnabledProcessor.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+/**
+ * An interface to mark DSL processor as Cache Enabled
+ */
+public interface CacheEnabledProcessor {
```

Comment:
If you are using the Processor API and one of our Rocks DB stores, then they implement `ForwardingStateStoreSupplier` and in these cases we don't want Caching.
So yes, we need something else to identify that we want caching enabled